snapcraft/tests/matchers.py

```diff
@@ -23,20 +23,29 @@ class HasArchitecture:
 
     def __init__(self, expected_arch):
         self._expected_arch = expected_arch
-        self._ms = magic.open(magic.NONE)
-        self._ms.load()
 
     def __str__(self):
         return 'HasArchitecture()'
 
     def match(self, file_path):
-        magic = self._ms.file(file_path)
-        # Catch exceptions on splitting the string to provide context
-        # This includes "cannot open `...' (No such file or directory)"
+        cmd = ['file', file_path]
         try:
-            arch = magic.split(',')[1]
+            output = subprocess.check_output(cmd).decode()
```

Comment:
Should you be specifying an encoding, here?

Reply:
The default is utf-8 and from what I've read is sane enough (and common practice)

Reply:
which we required btw due to use of click